Revit_Core_Engine/Query/Points.cs

```diff
@@ -0,0 +1,74 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2022, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using Autodesk.Revit.DB;
+using BH.oM.Base.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Returns control points of the CurveLoop.")]
+        [Input("curveLoop", "CurveLoop to get the points from.")]
+        [Output("points", "List of uniqe curve loop points")]
+        public static List<XYZ> Points(this CurveLoop curveLoop)
```

Comment:
I would call this method `Tessellate` and write it along these lines:
```suggestion
        public static List<XYZ> Tessellate(this CurveLoop curveLoop)
        {
            List<XYZ> result = new List<XYZ>();
            foreach (Curve curve in curveLoop)
            {
                IList<XYZ> points = curve.Tessellate();
                if (points.Count != 0)
                    result.AddRange(points.Take(points.Count - 1));
            }
            
            if (curveLoop.IsOpen())
                result.Add(curveLoop.Last().GetEndPoint(1));
            
            return result;
        }
```

Wouldn't that return the tessellated loop? I think this would be more correct than your solution because its order will not be distorted by culling duplicated (which can always be done separately if needed).

*I typed the above on the web page so the code may not compile, just showing the concept

Reply:
Good idea - to be done in the new branch.